registries/adapters/oauth/client.go

```diff
@@ -81,7 +81,16 @@ func (c *Client) NewRequest(path string) (*http.Request, error) {
 	if err != nil {
 		return nil, err
 	}
-	req.URL.Path = path
+
+	// path might be a fully qualified URL and we only want the path
+	// and query bits
+	thebits, err := url.Parse(path)
+	if err != nil {
+		return nil, err
+	}
+
+	req.URL.Path = thebits.Path
```

Comment:
could we just set `req.URL = thebits`?

Reply:
The comment on `NewRequest` states they use the base URL from the client. So I wanted to keep the same functionality. 
```golang
// NewRequest - creates and returns a *http.Request assuming the GET method.
// The base URL configured on the Client gets used with its Path component
// replaced by the path argument. If a token is available, it is added to the
// request automatically. "Accept: application/json" is added to all requests.
// The caller should customize the request as necessary before using it.
```